### PR TITLE
lib: add crls fn.

### PR DESCRIPTION
**Pull request by cpu**
Previously support for reading CRLs was added (https://github.com/rustls/pemfile/pull/20), but only exposed through the catch-all `read_all` function.

For use cases where one wants to _only_ read CRLs from a `.pem` it's helpful to have a `crls` function (ala the existing `certs` function).

This commit adds the `crls` function and a small test case.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -67,6 +67,23 @@ pub fn certs(rd: &mut dyn io::BufRead) -> Result<Vec<Vec<u8>>, io::Error> {
     }
 }
 
+/// Extract all the certificate revocation lists (CRLs) from `rd`, and return a vec of byte vecs
+/// containing the der-format contents.
+///
+/// This function does not fail if there are no CRLs in the file --
+/// it returns an empty vector.
+pub fn crls(rd: &mut dyn io::BufRead) -> Result<Vec<Vec<u8>>, io::Error> {
+    let mut crls = Vec::<Vec<u8>>::new();
```
Comment (djc) [inline src/lib.rs:76]: I don't think you need the turbofish here?

File: tests/data/crl.pem [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+BBIZIF50g20U+3izqpHutTD10itdk9TLsSceJHpwTkNJtaWMkOfBV28nKzEzVutV
+f6WzRpURGzui6nQy7aIqImeanpoBoz323psMfC32U0uMBCZltyHNqsX58/2Uhucx
+0IPnitNuhv4scCPf/jeRfGIWDrTf1/25LDzRxyg1S4z9aa+3GM4O3dqy4igZEhgT
+q3pjlJ2hUL5E0oqbZDIQD1SN8UUUv5N2AjwZcxVBNnYeGyuO7YpTBYiu62o73iL2
+CjgElfaMq/9hEr9GR9kJozh7VTxtQPbnr4DiucQvhv8o/A1z+zkC0gj8iCLFtDbO
+8bvDowcdle9LKkrLaBe6sO+fSH/I9Wj8vrEJKsuwaEraIdEaq2VrIMUPEWN0/MH9
+vTwHyadGSMK4CWtrn9fCAgSLw6NX74D7Cx1IaS8vstMjpeUqOS0dk5ThiW47HceB
+DTko7rV5N+RGH2nW1ynLoZKCJQqqZcLilFMyKPui3jifJnQlMFi54jGVgg/D6UQn
+7dA7wb2ux/1hSiaarp+mi7ncVOyByz6/WQP8mfc=
+-----END X509 CRL-----
```
Comment (djc) [inline tests/data/crl.pem:16]: Add a trailing newline?

**Review by djc — APPROVED** [review]

**Comment by cpu** [comment]
> cpu force-pushed the cpu-crl-support-pt2 branch from 4efa7bb to 1b50051 

* applied review feedback
* fixed an unrelated warning

Will look at why CI let that warning through separately.

**Comment by cpu** [comment]
> Will look at why CI let that warning through separately.

Very simple explanation. We weren't running `clippy` in CI at all. Fixed in https://github.com/rustls/pemfile/pull/22

**Pull request merged**